src/sentry/spans/buffer.py

```diff
@@ -519,13 +522,21 @@ def _load_segment_data(self, segment_keys: list[SegmentKey]) -> dict[SegmentKey,
             for key in segment_keys:
                 ingested_count_key = b"span-buf:ic:" + key
                 p.get(ingested_count_key)
+                ingested_byte_count_key = b"span-buf:ibc:" + key
+                p.get(ingested_byte_count_key)
 
-            ingested_count_results = p.execute()
+            ingested_results = p.execute()
 
         # Calculate dropped counts: total ingested - successfully loaded
-        for key, ingested_count in zip(segment_keys, ingested_count_results):
+        for i, key in enumerate(segment_keys):
+            ingested_count = ingested_results[i * 2]
+            ingested_byte_count = ingested_results[i * 2 + 1]
+
             if ingested_count:
                 total_ingested = int(ingested_count)
+                metrics.timing(
+                    "spans.buffer.flush_segments.ingested_spans_per_segment", total_ingested
+                )
                 successfully_loaded = len(payloads.get(key, []))
                 dropped = total_ingested - successfully_loaded
                 if dropped <= 0:
```

Comment:
i think this `continue` will prevent the `ingested_byte_count` from being emitted for non-dropped segments.